Examples/source/InstrumentAbstraction/ForceVoltageSMU.cs

```diff
@@ -19,88 +20,111 @@ namespace NationalInstruments.SemiconductorTestLibrary.Examples.InstrumentAbstra
+
+            smuPins.ForceVoltage(perPinLevelsToForce);
+        }
 
-            var pinASettings = new DCPowerSourceSettings { Level = 1.0, Limit = 0.09 };
-            var pinBSettings = new DCPowerSourceSettings { Level = 3.3, Limit = 0.01 };
-            var smuSettings = new Dictionary<string, DCPowerSourceSettings>()
+        /// <summary>
+        /// This example demonstrates how to force different voltage levels and current limits for different pins across all sites.
+        /// The example assumes there are exactly 2 pins.
+        /// </summary>
+        /// <param name="tsmContext">The <see cref="ISemiconductorModuleContext"/> object.</param>
+        internal static void DifferentValuesPerSmuPin(ISemiconductorModuleContext tsmContext)
+        {
+            var sessionManager = new TSMSessionManager(tsmContext);
+            // This dictionary is hard coded for demonstration purposes,
+            // but could otherwise be built by looping over array input parameters of the same length.
+            var perPinSmuSettings = new Dictionary<string, DCPowerSourceSettings>()
             {
-                [pinNames[0]] = pinASettings,
-                [pinNames[1]] = pinBSettings,
+                ["PinA"] = new DCPowerSourceSettings { Level = 1.0, Limit = 0.09 },
+                ["PinB"] = new DCPowerSourceSettings { Level = 3.3, Limit = 0.01 },
             };
 
-            smuPins.ForceVoltage(smuSettings);
+            var smuPins = sessionManager.DCPower(perPinSmuSettings.Keys.ToArray());
+
+            smuPins.ForceVoltage(perPinSmuSettings);
         }
 
         /// <summary>
         /// This example demonstrates how to configure different voltage levels for each site.
-        /// The example assumes there are 2 pins and 4 sites.
+        /// The example assumes there is a maximum of 4 sites.
         /// </summary>
         /// <param name="tsmContext">The <see cref="ISemiconductorModuleContext"/> object.</param>
-        internal static void DifferentValuesPerSiteSmu(ISemiconductorModuleContext tsmContext)
+        /// <param name="smuPinNames">The SMU pins to force voltage on</param>
+        internal static void DifferentValuesPerSiteSmu(ISemiconductorModuleContext tsmContext, string[] smuPinNames)
         {
-            var pinNames = new string[] { "PinA", "PinB" };
-            var perSiteVoltageLevelsArray = new double[] { 3.2, 3.3, 3.1, 3.8 };
             var sessionManager = new TSMSessionManager(tsmContext);
-            var smuPins = sessionManager.DCPower(pinNames);
-
-            var voltageLevelsSiteData = new SiteData<double>(perSiteVoltageLevelsArray);
+            // Max Site count of 4 expected
+            var perSiteVoltageLevelsArray = new double[] { 3.2, 3.3, 3.1, 3.8 };
+            var activeSites = tsmContext.SiteNumbers.ToArray();
+            // Array must be resized based on number of on active sites
+            Array.Resize(ref perSiteVoltageLevelsArray, activeSites.Length);
```

Comment:
If the activeSites is {1, 3}, would the resized array be {3.2, 3.3} or {3.3, 3.8}?

Reply:
@lornazh, good point, I am going to change this to the following, to be clearer:

```C#
var activeSites = tsmContext.SiteNumbers.ToArray();
// Unique per-site data, assuming a max site count of 4.
// Each index of the array represents the corresponding site number.
var perSiteVoltageLevelsArray = new double[] { 3.2, 3.3, 3.1, 3.8 };
// Note the site unique values must be filtered based on the active sites.
var activeSiteUniqueValues = new double[activeSites.Length];
for (int i = activeSites.Length - 1; i >= 0; i--)
{
    activeSiteUniqueValues[i] = perSiteVoltageLevelsArray[i];
}
```

I will also do the same for `DifferentValuesPerSiteAcrossAllPpmuPins` method in the ForceVoltagePPMU.cs file as well.

Reply:
@Mattjet27 when activeSites is {1, 3}, your new logic will reproduce {3.2, 3.3} right? Which, I think is the same behavior as Array.Resize? But my understanding is, we would like to produce {3.3, 3.8}, do you agree?

Below is a way how we produce {3.3, 3.8}.

var activeSiteUniqueValues = tsmContext.SiteNumbers.Select(siteNumber => perSiteVoltageLevelsArray[siteNumber]).ToArray();

Reply:
@lornazh, as discussed offline, I had meant for this to be the following:
```C#
for (int i = 0; i < activeSites.Length; i++)
{ 
   activeSiteUniqueValues[i] = perSiteVoltageLevelsArray[activeSites[i]];
}
```
I will raise a new PR to address this now.